Crash on experimentalObjectRestSpread

This plugin causes eslint to crash when the code contains object rest/spread properties ([a Stage 3 feature](https://github.com/tc39/proposal-object-rest-spread)) with the necessary parser flags enabled.

Command-line arguments:
```console
$ eslint.cmd --no-eslintrc --parser-options=ecmaFeatures:{experimentalObjectRestSpread:true},ecmaVersion:8 --plugin dollar-sign temp.js
```
Works as expected with this `temp.js`:
```js
/* eslint "dollar-sign/dollar-sign": "error" */
var foo = $('a');
```
```
D:\Code\temp.js
  2:5  error  jQuery identifiers must start with a $  dollar-sign/dollar-sign

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```
But crashes with this `temp.js`:
```js
/* eslint "dollar-sign/dollar-sign": "error" */
var foo = { ...{ a: 1 } };
```
```
Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at C:\Users\Username\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:152:41
    at Array.forEach (<anonymous>)
    at Linter.checkObjectExpression (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:149:9)
    at emitOne (events.js:115:13)
    at Linter.emit (events.js:210:7)
    at NodeEventGenerator.applySelector (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint\lib\util\node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint\lib\util\node-event-generator.js:294:22)
    at NodeEventGenerator.enterNode (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint\lib\util\node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:602:23)
    at Traverser.enter (C:\Users\Username\AppData\Roaming\npm\node_modules\eslint\lib\linter.js:925:36)
```
If you remove the eslint comment with the plugin rule from the file, elsint works correctly. 

I'm on Win7x64 with node.js v8.4.0, npm v5.3.0 and eslint v4.5.0, with eslint and dollar-sign installed globally.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,10 +1,9 @@
 sudo: false
 language: node_js
 node_js:
-  - "5"
-  - "4"
-  - "0.12"
-  - "0.10"
+  - 8
+  - 6
+  - 4
 script: npm test
 branches:
   only:
@@ -17,4 +16,4 @@ deploy:
     secure: b6CNngWFc9wP1WmAE8IBbigKvq7kZqi1LNV3bl8lSfvTv3Sm+QM4J3HHb2dNfvc5JZwMg0iiU53hhCFeuRzbt24/tmocIcaS0HHPhiMVwRecL4JnmCCxqhm6M1L3UNmjdgUW7wm0psDkufCcZPBxqB1NIW/7n0wTBe9MI99iAFuiiMF7tSS61AIs3iCJpnGSUjayAuOf++Addab03qpKaAB3Cl8vOa5akIttYMsEpEtWeWAayUt5wGOvHktCmPFe9WfZi7f3/4i1i85yzlRAqdSiTeyJ6+Nf/eJLK2eCUeNOK6ycWQk7M/KSqo0CHSuyRJtHi3iEpazxD4O3S8b2MrHjAOcQEmvY2A50FrQfpxvwaRcOWE91+S+41+CjlAHKx6IVJSOZpROFD8s8YCkqHBIO6TlFRJsgpTim9CH0fZ0Ec9auSYhihV/tNKBXFHjqcMudtRMi76Zvr+XwCNReiNpNOAn0Rd68KFwR2aAsx9rejpIaDQddCRtXpleQrpb0TM0cY0R3UKiJWfPyZ7+cjHn6OISAm/uHX/FjeXWRXS2yjDk5I3L1O20JaYwlMCRtITEL970J8lH4qRaUM5W7RJ/5lvscnSgBRj+CEmDcIFVbWFGKzWDLDNALE02PcInGknarMSKaOH5EWxGruqRB0gOhkAHvZQgOsgLIzhZMvgs=
   on:
     tags: true
-    node: 5
+    node: 8

--- a/package.json
+++ b/package.json
@@ -20,8 +20,8 @@
     "eslint": ">=1.0.0"
   },
   "devDependencies": {
-    "eslint": "^1.2.0",
-    "mocha": "^2.4.5"
+    "eslint": "^4.11.0",
+    "mocha": "^4.0.1"
   },
   "license": "MIT"
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-dollar-sign",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "Enforce $varName for jQuery assignment.",
   "keywords": [
     "eslint",

--- a/rules/dollar-sign.js
+++ b/rules/dollar-sign.js
@@ -149,6 +149,10 @@ module.exports = function(context) {
 		props.forEach(function(prop) {
 			var left = prop.key;
 
+			if (!left) {
+				return;
+			}
+
 			if (shouldVarNameStartWithDollar(left.name, prop.value)) {
 				reportIdentifier(left, false);
 			}

--- a/tests/dollar-sign.js
+++ b/tests/dollar-sign.js
@@ -129,6 +129,10 @@ ruleTester.run('dollar-sign', rule, {
 		'w.a = 1 + 2;',
 		// multi level object assignment
 		'a.b.$c = $()',
+		// object spread
+		{ code: 'var foo = { ...{ a: 1 } };', parserOptions: { ecmaFeatures: { experimentalObjectRestSpread: true } } },
+		// object rest
+		{ code: 'var { ...foo } = { a: 1 };', parserOptions: { ecmaFeatures: { experimentalObjectRestSpread: true } } },
 
 		//// option value `"ignoreProperties"`
 

--- a/tests/dollar-sign.js
+++ b/tests/dollar-sign.js
@@ -12,7 +12,7 @@ var RuleTester = require("eslint").RuleTester;
 
 var errorMessage = 'jQuery identifiers must start with a $';
 
-var ruleTester = new RuleTester();
+var ruleTester = new RuleTester({ env: { es6: true } });
 ruleTester.run('dollar-sign', rule, {
 	valid: [
 		// basic jquery operator with dollar
@@ -60,15 +60,15 @@ ruleTester.run('dollar-sign', rule, {
 		// jquery operator with dollar and single quotes around selector
 		'var $x = $(\'.foo\');',
 		// object destructuring
-		{ code: 'var {beep, boop} = meep;\nvar $s = $("#id")', ecmaFeatures: { destructuring: true } },
-		{ code: 'var {beep, boop} = $("#id")', ecmaFeatures: { destructuring: true } },
+		'var {beep, boop} = meep;\nvar $s = $("#id")',
+		'var {beep, boop} = $("#id")',
 		// object destructuring without var
-		{ code: '({beep, boop} = $("#id"))', ecmaFeatures: { destructuring: true } },
+		'({beep, boop} = $("#id"))',
 		// array destructuring
-		{ code: 'var [beep, boop] = meep;\nvar $s = $("#id")', ecmaFeatures: { destructuring: true } },
-		{ code: 'var [beep, boop] = $("#id")', ecmaFeatures: { destructuring: true } },
+		'var [beep, boop] = meep;\nvar $s = $("#id")',
+		'var [beep, boop] = $("#id")',
 		// array destructuring without var
-		{ code: '([beep, boop] = $("#id"))', ecmaFeatures: { destructuring: true } },
+		'([beep, boop] = $("#id"))',
 		// defined with a non-jQuery type
 		'var x = 5; x = $(".foo");',
 		// late assignment with jQuery
@@ -193,7 +193,6 @@ ruleTester.run('dollar-sign', rule, {
 		{
 			code: 'var {foo} = {foo: $(".foo")}',
 			output: 'var {foo} = {foo: $(".foo")}',
-			ecmaFeatures: { destructuring: true },
 			errors: [{
 				message: errorMessage,
 				type: 'Identifier',
@@ -370,7 +369,6 @@ ruleTester.run('dollar-sign', rule, {
 		{
 			code: 'var x = $(".foo"); ({ x });',
 			output: 'var x = $(".foo"); ({ x });',
-			ecmaFeatures: { objectLiteralShorthandProperties: true },
 			errors: [{
 				message: errorMessage,
 				type: 'Identifier',

